framework/auth/decorators.py

```diff
@@ -35,7 +39,7 @@ def wrapped(*args, **kwargs):
         if kwargs['auth'].logged_in:
             return func(*args, **kwargs)
         else:
-            return redirect('/login/?next={0}'.format(request.path))
+            return redirect(CasClient(settings.CAS_SERVER_URL).get_login_url(request.url))
```

Comment:
Since we'll pretty much always pass `settings.CAS_SERVER_URL` to `CasClient`, might be useful to have a helper function:

```python
def get_cas_client():
    return CasClient(settings.CAS_SERVER_URL)
```